src/@next/components/organisms/CartRow/CartRow.tsx

```diff
@@ -40,28 +40,33 @@ export const CartRow: React.FC<IProps> = ({
   attributes = [],
   onRemove,
 }: IProps) => {
-  const [tempQuantity, setTempQuantity] = useState<number | string>(quantity);
+  const [tempQuantity, setTempQuantity] = useState<string>(quantity.toString());
   const [isTooMuch, setIsTooMuch] = useState(false);
 
   const handleBlurQuantityInput = () => {
-    const newQuantity =
-      typeof tempQuantity === "number"
-        ? tempQuantity
-        : parseInt(tempQuantity, 10);
-    const notEnoughQuantity = isNaN(newQuantity) || newQuantity <= 0;
-    const tooMuchQuantity = newQuantity > maxQuantity;
-    if (notEnoughQuantity && !tooMuchQuantity) {
-      setTempQuantity(quantity);
-    } else if (!notEnoughQuantity && tooMuchQuantity) {
-      setTempQuantity(maxQuantity);
-      onQuantityChange(maxQuantity);
+    let newQuantity = parseInt(tempQuantity, 10);
+    let newTempQuantity = newQuantity.toString(); // To make newTempQuantity consistent with newQuantity
```

Comment:
Dont get the consistency comment. 
Also - having string and int near with so familiar names is asking for the troubles. Why not cast to string once in setTempQuantity method?

Reply:
Right, fixed